packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +290,119 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed'] &&
+        %w[nsupdate nsupdate_gss]
+          .include?(@scenario_answers['foreman_proxy']['dns_provider'])
+    end
+
+    def should_update_dns?
+      dns_managed? && !@options[:skip_dns]
+    end
+
+    def query_dns_server(fqdn, zone, nameserver_ip)
+      query_dns = Resolv::DNS.new(nameserver: [nameserver_ip], search: [], ndots: 1)
+      a_record = query_dns.getresource(fqdn, Resolv::DNS::Resource::IN::A)
+      ip = a_record.address.to_s if a_record
+      soa_record = query_dns.getresource(zone, Resolv::DNS::Resource::IN::SOA)
+      serial = soa_record.serial if soa_record
+      {
+        ip: ip,
+        serial: serial
+      }
+    end
+
+    def new_dns_values
+      new_vals = OpenStruct.new
+
+      new_vals.dns_server = @scenario_answers['foreman_proxy']['dns_server']
+      new_vals.zone = @scenario_answers['foreman_proxy']['dns_zone']
+      new_vals.reverse_zones = [@scenario_answers['foreman_proxy']['dns_reverse']].flatten
+      new_vals.soa_admin_domain = "root.#{new_vals.zone}"
+      new_vals.key_file = @scenario_answers['foreman_proxy']['keyfile']
+      new_vals.old_fqdn = @old_hostname
+      new_vals.new_fqdn = @new_hostname
+
+      ip_serial_data = query_dns_server(@old_hostname, new_vals.zone, new_vals.dns_server)
+      new_vals.ip = ip_serial_data[:ip]
+      serial = ip_serial_data[:serial]
+      new_vals.new_serial = serial + 1
+
+      new_vals.reverse_soa_records = []
+      new_vals.reverse_zones.each do |reverse_zone|
+        reverse_serial = query_dns_server(@old_hostname, reverse_zone, new_vals.dns_server)[:serial]
+        new_vals.reverse_soa_records << {
+          reverse_zone: reverse_zone,
+          new_reverse_serial: reverse_serial + 1
+        }
+      end
+      run_cmd('rndc thaw')
+
+      new_vals
+    end
+
+    def nsupdate_command(dns_entries, key_file)
+      "echo -e \"#{dns_entries}\" | nsupdate -l -k #{key_file}"
```

Comment:
Does this need to be updated to reflect the 'provider' @ekohl mentioned?

Reply:
@jturel I thought that provider value is only used to determine whether we should be touching DNS.  What would need to change about the nsupdate command?

Reply:
You might be correct. I asked since I don't know the answer

Reply:
The provider can be `nsupdate_gss` which means you need to add options to pass the tsig keytab and principal. You can also just say that the particular provider is not supported.

Reply:
@ekohl Thanks, after discussing with @chris1984 I feel like it will be best to only support the `nsupdate` provider for now.